trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/ConfigurationManagerService.java

```diff
@@ -155,9 +161,15 @@ public Response postConfiguration(@Parameter(description = CFG_TYPE_ID) @PathPar
             return Response.status(Status.BAD_REQUEST).entity(EndpointConstants.MISSING_PARAMETERS).build();
         }
 
-        @SuppressWarnings("null")
-        @NonNull Map<@NonNull String, @NonNull Object> params = queryParameters.getParameters();
-
+        @NonNull Map<@NonNull String, @NonNull Object> params = new HashMap<>();
+        if (configurationSource.getConfigurationSourceType().getSchemaFile() != null) {
+            // Pass JSON object as json string
+            params.put(TmfConfiguration.JSON_STRING_KEY, queryParameters.getParameters().toString());
+        } else {
+            // Convert it to a Map<String, Object>
```

Comment:
We can discuss to remove the support for the query param map in the TSP to avoid having clients to implement support for 2 types of user input. (To be discussed).